src/common/in_mem_overflow_buffer_utils.cpp

```diff
@@ -16,7 +16,7 @@ void InMemOverflowBufferUtils::copyString(
 }
 
 void InMemOverflowBufferUtils::copyListRecursiveIfNested(const ku_list_t& src, ku_list_t& dst,
```

Comment:
Not related to this PR, but I don't think we need this function anymore. Also `    template<typename T>
    static inline void setListElement(ku_list_t& result, uint64_t elementPos, T& element,
        const DataType& dataType, InMemOverflowBuffer& inMemOverflowBuffer)`.
Can you double check?
We should also move 
```
    static inline void allocateSpaceForList(
        ku_list_t& list, uint64_t numBytes, InMemOverflowBuffer& buffer) {
        list.overflowPtr = reinterpret_cast<uint64_t>(buffer.allocateSpace(numBytes));
    }
```
to ValueVectorUtils.

Reply:
Removed these functions as well as the `allocateSpaceForList`